tests/Unit/NumericalAlgorithms/Interpolation/Test_IrregularInterpolant.cpp

```diff
@@ -230,6 +242,145 @@ void test_irregular_interpolant_mixed_quadrature() {
   }
 }
 
+template <size_t Dim>
+Domain<Dim> create_domain(double length, size_t n_pts) noexcept;
+
+template <>
+Domain<3> create_domain<3>(const double length, const size_t n_pts) noexcept {
+  const domain::creators::Brick creator{{{0.0, 0.0, 0.0}},
+                                        {{length, length, length}},
+                                        {{0, 0, 0}},
+                                        {{n_pts, n_pts, n_pts}},
+                                        {{false, false, false}}};
+  return creator.create_domain();
+}
+
+template <>
+Domain<2> create_domain<2>(const double length, const size_t n_pts) noexcept {
+  const domain::creators::Rectangle creator{{{0.0, 0.0}},
+                                            {{length, length}},
+                                            {{0, 0}},
+                                            {{n_pts, n_pts}},
+                                            {{false, false}}};
+  return creator.create_domain();
+}
+
+template <>
+Domain<1> create_domain<1>(const double length, const size_t n_pts) noexcept {
+  const domain::creators::Interval creator{{{0.0}},   {{length}}, {{0}},
+                                           {{n_pts}}, {{false}},  nullptr};
+  return creator.create_domain();
+}
+
+template <size_t Dim>
+tnsr::I<DataVector, Dim, Frame::Logical> create_target_points(
+    size_t n_random_target_points) {
+  tnsr::I<DataVector, Dim, Frame::Logical> xi_target{n_random_target_points + 2,
+                                                     -1.0};
+  for (size_t d = 0; d < Dim; ++d) {
+    xi_target.get(d)[1] = 1.0;
+  }
+  return xi_target;
+}
+
+namespace Tags {
+struct Scalar : ::db::SimpleTag {
+  using type = ::Scalar<DataVector>;
+};
+}  // namespace Tags
+
+using var_tags = tmpl::list<Tags::Scalar>;
+
+template <size_t Dim>
+Variables<var_tags> polynomial(
+    const tnsr::I<DataVector, Dim, Frame::Inertial>& x, const size_t order) {
```

Comment:
can you specify the degree instead of the order? Unfortunately different sources (text books, papers) define the order of a polynomial differently, while degree is unambiguous